dynamic/main.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/blang/semver"
+	"github.com/opentofu/opentofu/shim"
+	"github.com/pulumi/pulumi-terraform-bridge/pf/proto"
+	pfbridge "github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+)
+
+const (
+	name    = "terraform-bridge"
+	version = "0.0.1"
+)
+
+func initialSetup() (tfbridge.ProviderInfo, pfbridge.ProviderMetadata, func() error) {
+
+	var tfServer shim.Provider
+	defaultInfo := tfbridge.ProviderInfo{
+		DisplayName:  "Any Terraform Provider",
+		P:            proto.Empty(),
+		Name:         name,
+		Version:      version,
+		Description:  "A Pulumi provider for dynamically bridging terraform provdiers.",
+		MetadataInfo: &tfbridge.MetadataInfo{Path: "", Data: tfbridge.ProviderMetadata(nil)},
+	}
+
+	metadata := pfbridge.ProviderMetadata{
+		XGetSchema: func(ctx context.Context, req plugin.GetSchemaRequest) ([]byte, error) {
+
+			info := defaultInfo
+			if tfServer != nil {
+				info = providerInfo(ctx, tfServer)
+			}
+
+			packageSchema, err := tfgen.GenerateSchemaWithOptions(tfgen.GenerateSchemaOptions{
+				ProviderInfo: info,
+				DiagnosticsSink: diag.DefaultSink(io.Discard, os.Stderr, diag.FormatOptions{
```

Comment:
Can you actually log to stderr from providers safely?

Should stdout be redirected somewhere other than discard?

Reply:
I couldn't when I generated the schema promptly, but I can now that this is correctly part of Parameterize.